backend/functions/src/emailing/functions.ts

```diff
@@ -1,4 +1,50 @@
 import axios from 'axios'
+
+// firebase imports
+import admin from 'firebase-admin'
+import { db } from '../config'
+const courseRef = db.collection('courses')
+
+// ==== Timestamp helper functions
+
+type timestampsType = { [key: string]: string }
+
+const timestamps: timestampsType = {
+  'Share matched results': 'shareMatchEmailTimestamp',
+  'First no match notification': 'checkInEmailTimestamp',
+  'Second no match notification': 'firstNoMatchEmailTimestamp',
+  'Request to add student to group': 'secondNoMatchEmailTimestamp',
+  'Request to add student to group (late)': 'addStudentEmailTimestamp',
+  'Ask to join group': 'lateAddStudentEmailTimestamp',
+  'Check in with groups': 'askJoinGroupEmailTimestamp',
+}
```

Comment:
I think some of these are not quite right, e.g. `'Second no match notification': 'firstNoMatchEmailTimestamp'`

Reply:
Oop yeah I think I didn't match them up correctly when I was reformatting it lol. good catch